### PR TITLE
Avoid showing `[object Object]` in status error

### PR DESCRIPTION
**Pull request by huayang-codaio**
Alternative is to show JSON.stringify(body). 

however it may not be well structured for display purpose. For example Coda API 401 looks like 

image

So let's simply show this is a fetcher status error with code.

### CODE REVIEW

**Review by spencer-codaio — COMMENTED** [review]
hmm my preference is towards being more verbose / show the full body to have more info available. Or is your thinking we need some more work on handling long error messages in the debug log to show this?

**Comment by huayang-codaio** [comment]
That makes sense. Codans can see the errors from system logs. but indeed external developers don't have any info.

**Review by spencer-codaio — APPROVED** [review]

File: api.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class StatusCodeError extends Error {
   options: FetchRequest;
   response: StatusCodeErrorResponse;
   constructor(statusCode: number, body: any, options: FetchRequest, response: StatusCodeErrorResponse) {
-    super(`${statusCode} - ${body}`);
+    super(`Fetch failed with status code: ${statusCode} - ${JSON.stringify(body, null, 4)}`);
```
Comment (jonathan-codaio) [inline api.ts:80]: I think `util.inspect()` is preferable. Expands a full object into a string, without transforming it.

Comment (huayang-codaio) [reply api.ts:80]: util is a node library, while this is used in thunk, non-browserified. I don't want to inject `util` just for this.

Comment (jonathan-codaio) [reply api.ts:80]: Ah ok, makes sense, too bad.

**Pull request merged**